benchmark/benchresult/main.go

```diff
@@ -0,0 +1,134 @@
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	for _, d := range data {
+		m[d.RunMode+"-"+d.Features.String()] = d
+	}
+}
+
+func intChange(title string, val1, val2 int64) string {
+	percentChange := strconv.FormatFloat(float64((val2-val1)*100/val1), 'f', 2, 64) + "% "
+	return fmt.Sprintf("%10s  %12s  %12s   %8s \n", title, strconv.FormatInt(val1, 10),
+		strconv.FormatInt(val2, 10), percentChange)
+}
+
+func timeChange(title int, val1, val2 time.Duration) string {
+	percentChange := strconv.FormatFloat(float64((val2-val1)*100/val1), 'f', 2, 64) + "% "
+	return fmt.Sprintf("%10s  %12s  %12s   %8s \n", strconv.Itoa(title)+" latency", val1.String(),
+		val2.String(), percentChange)
+}
+
+func compareTwoMap(m1, m2 map[string]stats.BenchResults) {
+	for k2, v2 := range m2 {
+		if v1, ok := m1[k2]; ok {
+			changes := k2 + "-"
+			changes += fmt.Sprintf("%10s  %12s  %12s   %8s \n", "\nTitle", "\tBefore", "After", "\tPercentage")
+			changes += intChange("Bytes/op", v1.AllocedBytesPerOp, v2.AllocedBytesPerOp)
+			changes += intChange("Allocs/op", v1.AllocsPerOp, v2.AllocsPerOp)
+			changes += timeChange(v1.Latency[1].Percent, v1.Latency[1].Value, v2.Latency[1].Value)
+			changes += timeChange(v1.Latency[2].Percent, v1.Latency[2].Value, v2.Latency[2].Value)
+			fmt.Printf("%s\n", changes)
+		}
+	}
+}
+
+func compareBenchmark(file1, file2 string) {
+	var BenchValueFile1 map[string]stats.BenchResults
+	var BenchValueFile2 map[string]stats.BenchResults
+	BenchValueFile1 = make(map[string]stats.BenchResults)
+	BenchValueFile2 = make(map[string]stats.BenchResults)
+
+	createMap(file1, BenchValueFile1)
+	createMap(file2, BenchValueFile2)
+
+	compareTwoMap(BenchValueFile1, BenchValueFile2)
+}
+
+func printline(benchName, ltc50, ltc90, allocByte, allocsOp string) {
+	fmt.Printf("%-80s%12s%12s%12s%12s\n", benchName, ltc50, ltc90, allocByte, allocsOp)
+}
+func formatBenchmark(fileName string) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Fatalf("Read file %s error: %s\n", fileName, err)
+	}
+	defer f.Close()
+	var data []stats.BenchResults
+	decoder := gob.NewDecoder(f)
+	if err = decoder.Decode(&data); err != nil {
+		log.Fatalf("Decode file %s error: %s\n", fileName, err)
+	}
+	if len(data) == 0 {
+		log.Fatalf("No data in file %s\n", fileName)
+	}
+	printPos := data[0].SharedPosion
+	fmt.Println("\n Shared features: \n" + strings.Repeat("-", 20))
+	fmt.Print(stats.PartialPrintString(printPos, data[0].Features, true))
+	fmt.Println(strings.Repeat("-", 35))
+	for i := 0; i < len(data[0].SharedPosion); i++ {
+		printPos[i] = !printPos[i]
+	}
+	printline("Name", "latency-50", "latency-90", "Alloc (B)", "Alloc (#)")
+	for _, d := range data {
+		name := d.RunMode + stats.PartialPrintString(printPos, d.Features, false)
+		printline(name, d.Latency[1].Value.String(), d.Latency[2].Value.String(),
+			strconv.FormatInt(d.AllocedBytesPerOp, 10), strconv.FormatInt(d.AllocsPerOp, 10))
```

Comment:
make printline() take interface{} instead of string and you can remove the `strconv.FormatInt`s and the `.String()` calls too.  (Print using `%v` instead of `%s`.)

Reply:
Done. printline receives interface and all printed as %v.